modules/rtdModule/index.js

```diff
@@ -159,130 +180,153 @@ export function init(config) {
  */
 function setEventsListeners() {
   events.on(CONSTANTS.EVENTS.AUCTION_INIT, (args) => {
-    subModules.forEach(sm => { sm.auctionInit && sm.auctionInit(args, sm.config) })
+    subModules.forEach(sm => { sm.onAuctionInitEvent && sm.onAuctionInitEvent(args, sm.config, _userConsent) })
   });
   events.on(CONSTANTS.EVENTS.AUCTION_END, (args) => {
-    subModules.forEach(sm => { sm.auctionEnd && sm.auctionEnd(args, sm.config) })
-  });
-  events.on(CONSTANTS.EVENTS.BEFORE_REQUEST_BIDS, (args) => {
-    subModules.forEach(sm => { sm.updateBidRequest && sm.updateBidRequest(args, sm.config) })
+    getAdUnitTargeting(args);
+    subModules.forEach(sm => { sm.onAuctionEndEvent && sm.onAuctionEndEvent(args, sm.config, _userConsent) })
   });
   events.on(CONSTANTS.EVENTS.BID_RESPONSE, (args) => {
-    subModules.forEach(sm => { sm.updateBidResponse && sm.updateBidResponse(args, sm.config) })
+    subModules.forEach(sm => { sm.onBidResponseEvent && sm.onBidResponseEvent(args, sm.config, _userConsent) })
   });
 }
 
 /**
- * get data from sub module
- * @param {AdUnit[]} adUnits received from auction
- * @param {function} callback callback function on data received
+ * loop through configured data providers If the data provider has registered getBidRequestData,
+ * call it, providing reqBidsConfigObj, consent data and module params
+ * this allows submodules to modify bidders
+ * @param {Object} reqBidsConfigObj required; This is the same param that's used in pbjs.requestBids.
+ * @param {function} fn required; The next function in the chain, used by hook.js
  */
-export function getProviderData(adUnits, callback) {
-  /**
-   * invoke callback if one of the conditions met:
-   * timeout reached
-   * all submodules answered
-   * all sub modules configured "waitForIt:true" answered (as long as there is at least one configured)
-   */
-
-  const waitForSubModulesLength = subModules.filter(sm => sm.config && sm.config.waitForIt).length;
-  let callbacksExpected = waitForSubModulesLength || subModules.length;
-  const shouldWaitForAllSubModules = waitForSubModulesLength === 0;
-  let dataReceived = {};
-  let processDone = false;
-  const dataWaitTimeout = setTimeout(done, _moduleConfig.auctionDelay || _moduleConfig.timeout || DEF_TIMEOUT);
+export function setBidRequestsData(fn, reqBidsConfigObj) {
+  _userConsent = getConsentData();
+
+  const relevantSubModules = [];
+  const prioritySubModules = [];
   subModules.forEach(sm => {
-    sm.getData(adUnits, onDataReceived.bind(sm));
+    if (typeof sm.getBidRequestData !== 'function') {
+      return;
+    }
+    relevantSubModules.push(sm);
+    const config = sm.config;
+    if (config && config.waitForIt) {
+      prioritySubModules.push(sm);
+    }
   });
 
-  function onDataReceived(data) {
-    if (processDone) {
-      return
+  const shouldDelayAuction = prioritySubModules.length && _moduleConfig.auctionDelay && _moduleConfig.auctionDelay > 0;
```

Comment:
can we rename this param to `auctionDelayMs` (assuming it is in MS)

Reply:
@mkendall07 - that would be a good idea, but it would be a breaking change for publishers -- some are already using the auctionDelay option and would have to change it. This could be a candidate for 6.0